server/ServerFSM.cpp

```diff
@@ -1744,13 +1752,18 @@ sc::result PlayingGame::react(const PlayerChat& msg) {
     int receiver;
     ExtractPlayerChatMessageData(message, receiver, data);
 
+    boost::posix_time::ptime timestamp = boost::posix_time::second_clock::universal_time();
+
+    server.PushChatMessage(timestamp, sender->PlayerName() + ": " + data);
```

Comment:
this formatting of `Name: message` should probably be done client-side, in case different languages have different layouts for that...

Reply:
Currently client doesn't support different layouts for that.
Also I have no idea about implementation. History message cann't have an player id because player could disconnect before message will be shown so player name is mandatory. But chat should have an player id as it used for coloring.
I think it better to add new message type for chat history.